test/spec/directives/bsSwitchSpec.js

```diff
@@ -672,23 +677,53 @@ describe('Directive: bsSwitch', function () {
   it('should watch updates in getterSetter', inject(makeTestGetterSetter()));
   it('should watch updates in getterSetter', inject(makeTestGetterSetter(true)));
 
-  function makeTestChange(input) {
+  function makeTestViewNgChange(input) {
     return function () {
-      compileDirective('change', input);
+      var element = compileDirective('ngChange', input);
+      scope.ngChange = jasmine.createSpy();
 
-      scope.change = function() {};
-      spyOn(scope, 'change');
+      // On - model change
+      scope.ngChange.calls.reset();
+      scope.model = true;
+      scope.$apply();
+      expect(scope.ngChange).not.toHaveBeenCalled();
 
-      scope.model = false;
+      // Indeterminate - model change
+      scope.ngChange.calls.reset();
+      scope.model = undefined;
       scope.$apply();
-      expect(scope.change).not.toHaveBeenCalled();
+      expect(scope.ngChange).not.toHaveBeenCalled();
+
+      // Off - view change
+      scope.ngChange.calls.reset();
```

Comment:
I would remove all of these `reset`s since the ngChange won't be called while this gives the impression that it may.